File: src/main/java/com/bloodbridge/service/BloodRequestService.java
package com.bloodbridge.service;

import com.bloodbridge.dto.bloodRequest.BloodRequestCreateRequest;
import com.bloodbridge.dto.bloodRequest.BloodRequestResponse;
import com.bloodbridge.dto.bloodRequest.BloodRequestUpdateRequest;

import java.util.List;

public interface BloodRequestService {
    BloodRequestResponse createBloodRequest(BloodRequestCreateRequest request);

    BloodRequestResponse getBloodRequestById(Long id);

    List<BloodRequestResponse> getAllBloodRequests();

    List<BloodRequestResponse> getMyBloodRequests();

    void deleteBloodRequest(Long id);

    BloodRequestResponse updateBloodRequest(Long bloodRequestId, BloodRequestUpdateRequest request);
}

File: bloodbridge-frontend/src/services/bloodRequestService.js
import API from "./api";

export const getAllBloodRequests = async () => {
    const response = await API.get("/blood-requests");
    return response.data;
};

export const getBloodRequestById = async (id) => {
    const response = await API.get(`/blood-requests/${id}`);
    return response.data;
};

export const createBloodRequest = async (request) => {
    const response = await API.post("/blood-requests", request);
    return response.data;
}

export const getMyBloodRequests = async () => {
    const response = await API.get("/blood-requests/my");
    return response.data;
}
